src/zope/i18nmessageid/message.py

```diff
@@ -31,18 +28,37 @@ class Message(unicode):
     message id itself implicitly serves as the default text.
     """
 
-    __slots__ = ('domain', 'default', 'mapping', '_readonly')
+    __slots__ = (
+        'domain', 'default', 'mapping', '_readonly',
+        'msgid_plural', 'default_plural', 'number')
 
-    def __new__(cls, ustr, domain=None, default=None, mapping=None):
-        self = unicode.__new__(cls, ustr)
+    def __new__(cls, ustr, domain=None, default=None, mapping=None,
+                msgid_plural=None, default_plural=None, number=None):
+        self = six.text_type.__new__(cls, ustr)
         if isinstance(ustr, self.__class__):
             domain = ustr.domain and ustr.domain[:] or domain
             default = ustr.default and ustr.default[:] or default
             mapping = ustr.mapping and ustr.mapping.copy() or mapping
```

Comment:
Pre-existing bug: this doesn't match the C logic.  AFAICT

    m1 = Message("blah", domain="d1")
    m2 = Message(m1, domain="d2")
    print(m2.domain)

will print d2 if we use the C implementation of Message, and it will print d1 if we use the Python implementation.

I think the C implementation is right and the Python one is wrong.